### DISCUSSION TITLE
Setting random access with dzsave from command line

### DISCUSSION THREAD

**Discussion by stecca21** [libvips/libvips#2916] [Q&A] [answered]
I'm trying to incorporate proper rotation into a dzsave command from the command line and can't figure out what I need to do. My test image is a jpeg with a metadata setting of "orientation=Rotate 90 degrees". Using the following command I get errors:

`vips dzsave <infile> <outfolder> --angle=d90`

I understand now that this is due to sequential reading in of the infile, but I can't find documentation on how to set access to random. Any help would be much appreciated.

**Comment by jcupitt** [comment, marked as answer]
Hi @stecca21,

The JPEG loader has an autorotate option which might do what you need:

```
vips dzsave testfile.jpg[autorotate] my_output_folder
```

It might need a huge amount of memory and/or disc though (I expect you realise).

**Comment by stecca21** [comment]
That does seem to work, so thanks for the quick response.

Regarding your comment about memory/disk usage - would you recommend approaching this with a different command or set of commands?

**Reply by jcupitt** [reply]
I don't think there's much you can do -- 90 degree rotate on a large file will always need *a lot* of memory or disc :(

**Reply by stecca21** [reply]
Gotcha - again, thanks for the quick response!
